Source/AssetRipper.Processing/AnimatorControllers/VirtualAnimationFactory.cs

```diff
@@ -410,12 +416,13 @@ private static IAnimatorState CreateAnimatorState(ProcessedAssetCollection virtu
 			for (int i = 0; i < Transition.ConditionConstantArray.Count; i++)
 			{
 				ConditionConstant conditionConstant = Transition.ConditionConstantArray[i].Data;
-				if (conditionConstant.ConditionMode != (int)AnimatorConditionMode.ExitTime)
+				if (!Transition.Has_ExitTime() || conditionConstant.ConditionMode != (int)AnimatorConditionMode.ExitTime)
```

Comment:
Why would `Transition.Has_ExitTime` have any relevance here?

Reply:
The `ExitTime` of transitions on Unity 4 is stored as a condition.
`Transition.Has_ExitTime` is the same as `generatedStateMachine.Has_ChildStates_C1107()` above, and is used to detect whether this transition is in Unity 5+ format.
However, in transitions on Unity 5+, conditions will not contain ExitTime conditions, so this condition mode judgment can be removed.

Reply:
> `Transition.Has_ExitTime` is the same as `generatedStateMachine.Has_ChildStates_C1107()`

They are not the same. They are from two different assets, which could have different Unity versions. `Transition` is from an `IAnimatorController` asset. You should use `!animatorStateTransition.Has_ExitTime_C1101()`.